src/controllers/v1/_common/demand.ts

```diff
@@ -47,7 +48,17 @@ export class DemandController extends Controller {
       throw new BadRequest('Attachment not found')
     }
 
-    const res = self || (await this.identity.getMemberBySelf(getAuthorization(req)))
+    // So self should be whoever is actually making this transaction -> which is Dave if there is a PROXY
+    // He is also the one who is associated with it in a db
+    let res: {
+      address: string
+      alias: string
+    }
+    if (env.PROXY !== null) {
+      res = self || (await this.identity.getMemberByAddress(env.PROXY, getAuthorization(req)))
+    } else {
+      res = self || (await this.identity.getMemberBySelf(getAuthorization(req)))
+    }
```

Comment:
This logic is repeated in several places. Can we move the entire of this to a new shared function that takes an `Identity` an `Env` and the authorization and does this logic and the relevant caching? That will allow you to deduplicate this logic and unit test it independantly